src/Process/ExecutionLoop.php

```diff
@@ -175,11 +216,25 @@ protected function dequeueDependentProcesses(): void
         foreach ($queued as $testClass => $processWrapper) {
             $delaySeconds = $processWrapper->getDelayMinutes() * 60;
 
-            if (in_array($processWrapper->getDelayAfter(), $doneClasses, true)
+            if (!$this->parallelLimitReached() && in_array($processWrapper->getDelayAfter(), $doneClasses, true)
                 && (time() - $done[$processWrapper->getDelayAfter()]->getFinishedTime()) > $delaySeconds
             ) {
                 if ($this->io->isVeryVerbose()) {
-                    $this->io->runStatus(sprintf('Unqueing testcase "%s"', $testClass));
+                    $this->io->runStatus(sprintf('Dequeing testcase "%s"', $testClass));
+                }
+                $processWrapper->setStatus(ProcessWrapper::PROCESS_STATUS_PREPARED);
+            }
+        }
+    }
+
+    protected function dequeueParallelProcesses(): void
+    {
+        $queued = $this->processSet->get(ProcessWrapper::PROCESS_STATUS_QUEUED);
+
+        foreach ($queued as $testClass => $processWrapper) {
+            if (!$processWrapper->isDelayed() && !$this->parallelLimitReached()) {
+                if ($this->io->isVeryVerbose()) {
+                    $this->io->runStatus(sprintf('Dequeing testcase "%s" from parallel limit', $testClass));
```

Comment:
`Dequeing testcase "%s" which was queued due to parallel limit`?

Reply:
`... which was queued because of parallel limit`